### ISSUE TITLE
IPython.display.IFrame running on remote JupyterLab references localhost on client machine

### ISSUE THREAD

**Issue by jcaxle** [ipython/ipython#14232]
I have a localhost server running on JupyterLab serving a web application which I would like to display on JupyterLabs on a client machine. However, when calling IPython.display.IFrame() on the client, it references localhost on the client machine instead of the remote server. Here is an example where http://localhost:57967 is where the web application is being served on the JupyterLabs server:

Calling IPython.display.IFrame() displays "refused to connect" error page.
![image](https://github.com/ipython/ipython/assets/145499292/f36205c0-fd14-4f12-98e0-81a35e84b5de)

However, http://localhost:57967 serves a web application and is accessible from the remote server
![image](https://github.com/ipython/ipython/assets/145499292/94b04696-690d-4cdb-88fe-47c734531526)

On the client machine, I have a web application running on http://localhost:58541 which is accessible by IPython.display.IFrame(). 
![image](https://github.com/ipython/ipython/assets/145499292/f1429374-672e-4ac6-9b4d-ff9f57402f2a)

However, http://localhost:58541 cannot be accessed from the remote server.
![image](https://github.com/ipython/ipython/assets/145499292/bc2c6e42-562b-4f0f-8ba7-a7331d11d3e4)

In summary, how would I be able to display http://localhost:57967 which is a web application served on the JupyterLab's server?

**Comment by Carreau** [comment]
That's not an IPython or JupyterLab thing, this is a how internet works.

`localhost` always refers to "the current machine". There is not much you can do in IPython. 
You need to get the real (external) IP or a domain name. 

This goes way beyond something we can solve in an issue and what you are trying to achieve.

One thing you can try, which is not scalable would be to use a service [like ngrok](https://www.softwaretestinghelp.com/ngrok-alternatives/), it will "give" you something that is a bit more persistant and accessible from the outside.

**Comment by ntroutman** [comment]
I've run into this as well and it is annyoing (but understandable) that an iframe is being rendered locally instead of on the server. Basically when we run:
```shell
! curl http://localhost:8080
```

It will execute the curl on the remote host, so the "localhost" there is the remote server, and things work as expected.

However, when running:
```python
from IPython.display import IFrame
IFrame('http://localhost:8080/status', width=700, height=350)
```

It ends up with the local webbrowser trying to resolve the IFrame, instead of the the notebook resolving and piping back the HTML. This would need something like a "remote iframe" where all the handling was done on the server and just the HTML was passed back for local rendering.
